### PR TITLE
[2/3] fix dsv3 awq issue 

### PR DESCRIPTION
**Pull request by AniZpZ**
## Motivation

related issue: https://github.com/sgl-project/sglang/issues/4462
The dsv3 awq issue requires a few PRs to fix, this is a part of it.
Currently dsv3 awq suffers speed problem because lacking of efficient moe w4a16 kernel

Author: @laixinn @huangtingwei9988

## Modifications
Add moe wna16 kernel to sglang

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/moe_wna16.py [hunk 1]
```diff
@@ -0,0 +1,511 @@
+# Adapted from https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/quantization/moe_wna16.py
+
+import logging
+from typing import Any, Callable, Dict, List, Optional
+
+import torch
+from vllm.model_executor.layers.quantization.awq import AWQConfig
```
Comment (zhyncs) [inline python/sglang/srt/layers/quantization/moe_wna16.py]: Please don't introduce vllm dependency

Comment (AniZpZ) [reply python/sglang/srt/layers/quantization/moe_wna16.py]: this could be removed after AWQConfig is moved into sglang
https://github.com/sgl-project/sglang/pull/4719

**Comment by zhyncs** [comment]
@AniZpZ https://github.com/sgl-project/sglang/pull/4719 has been merged. May you rebase the latest main and remove the vllm dependency

**Comment by AniZpZ** [comment]
> @AniZpZ #4719 has been merged. May you rebase the latest main and remove the vllm dependency

vllm dependency has been removed now. Do you have any other suggestions for the PR？

File: benchmark/deepseek_v3/README.md [hunk 2]
```diff
@@ -178,10 +178,12 @@ python3 -m sglang.bench_one_batch_server --model None --base-url http://10.0.0.1
 
 ### Example: Serving with 8 A100/A800 with AWQ Quantization
 
-AWQ does not support BF16, so add the `--dtype half` flag if AWQ is used for quantization. One example is as follows:
+AWQ does not support BF16, so add the `--dtype half` flag if AWQ is used for quantization.
```
Comment (zhyncs) [inline benchmark/deepseek_v3/README.md]: `AWQ does not support BF16` May you update this

**Comment by zhyncs** [comment]
@AniZpZ @laixinn Can you share the accuracy and performance benchmark results with the commands? Thanks.

File: benchmark/deepseek_v3/README.md (hunk 2 above)
Comment (AniZpZ) [reply benchmark/deepseek_v3/README.md]: fixed

**Comment by AniZpZ** [comment]
> @AniZpZ @laixinn Can you share the accuracy and performance benchmark results with the commands? Thanks.

The accuracy can be referred to https://github.com/sgl-project/sglang/issues/4462#issuecomment-2742996720 with minimal drop in accuracy on GSM8k and MMLU.

Model | MMLU | GSM8K
-- | -- | --
cognitivecomputations/DeepSeek-R1-AWQ | 86.8 | 94.7
cognitivecomputations/DeepSeek-V3-AWQ | 87.6 | 93.4
casperhansen/deepseek-coder-v2-instruct-awq | 77.5 | 92.8

And I benchmarked AWQ and origin FP8 with cognitivecomputations/DeepSeek-R1-AWQ in H20 * 8. The data is listed as follows. No optimization option enabled except for quantization.
The original FP8 requires fewer compute resources, while AWQ benefits from more KV blocks on memory-limited platforms, such as H20 * 8

input 4096 output 1024
BS | DeepSeek-R1-AWQ |   | DeepSeek-R1 |  
-- | -- | -- | -- | --
  | output throughput | throughput | output throughput | throughput
1 | 21.87 | 109.33 | 23.08 | 115.41
2 | 37.38 | 186.91 | 56.55 | 282.74
4 | 61.22 | 306.10 | 104.27 | 521.34
8 | 137.11 | 685.54 | 141.67 | 708.36
16 | 217.20 | 1086.00 | 69.55 | 347.74(kv block bounding)
32 | 244.62 | 1223.08 | 147.94 | 739.69

input 512 output 512

BS | DeepSeek-R1-AWQ |   | DeepSeek-R1 |  
-- | -- | -- | -- | --
  | output throughput | throughput | output throughput | throughput
1 | 24.14 | 48.28 | 29.97 | 59.93
2 | 40.92 | 81.83 | 64.82 | 129.65
4 | 64.69 | 129.37 | 69.15 | 138.31
8 | 190.33 | 380.66 | 181.82 | 363.65
16 | 313.96 | 627.92 | 293.14 | 586.29
32 | 503.29 | 1006.58 | 450.81 | 901.62
